scene/resources/packed_scene.cpp

```diff
@@ -314,6 +314,16 @@ Node *SceneState::instantiate(GenEditState p_edit_state) const {
 					ERR_FAIL_INDEX_V(nprops[j].value, prop_count, nullptr);
 
 					if (nprops[j].name & FLAG_PATH_PROPERTY_IS_NODE) {
+						if (node->get_scene_instance_load_placeholder()) {
```

Comment:
I think you may need to change this to something like, since otherwise direct node references get lost upon reloading a scene in the editor:
`if (!Engine::get_singleton()->is_editor_hint() && node->get_scene_instance_load_placeholder()) {`